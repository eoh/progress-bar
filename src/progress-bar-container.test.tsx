import * as React from 'react';
import { configure } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import { shallow, ShallowWrapper } from 'enzyme';
import ProgressBarContainer, { ProgressBarContainerProps } from './progress-bar-container';

describe('ProgressBarContainer', () => {
  let component: ShallowWrapper<ProgressBarContainer, {}>;
  let onIncrementClickCallback: jest.Mock<void>;
  let onDecrementClickCallback: jest.Mock<void>;

  const getProps = (overrides?: Partial<ProgressBarContainerProps>) => {
    onIncrementClickCallback = jest.fn();
    onDecrementClickCallback = jest.fn();

    const defaultProps = {
      min: 100,
      max: 100,
      onIncrementClick: onIncrementClickCallback,
      onDecrementClick: onDecrementClickCallback,
      progress: 50,
      fillColor: '#ccc',
      height: 50,
      width: 400,
    };

    return {
      ...defaultProps,
      ...overrides,
    };
  };

  describe('when initialized', () => {
    let props: ProgressBarContainerProps;

    beforeEach(() => {
      props = getProps();
      component = shallow(<ProgressBarContainer {...props} />);
    });

    it('matches snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
