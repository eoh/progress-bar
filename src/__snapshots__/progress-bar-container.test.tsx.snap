// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBarContainer when initialized matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProgressBarContainer
    fillColor="#ccc"
    height={50}
    max={100}
    min={100}
    onDecrementClick={[Function]}
    onIncrementClick={[Function]}
    progress={50}
    width={400}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ProgressBar
        bgColor={undefined}
        borderRadius={undefined}
        borderStyle={undefined}
        demo={undefined}
        fillColor="#ccc"
        height={50}
        max={100}
        min={100}
        onDecrement={[Function]}
        onIncrement={[Function]}
        progress={0}
        transition={undefined}
        width={400}
/>,
      "className": "progress-bar-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "bgColor": undefined,
        "borderRadius": undefined,
        "borderStyle": undefined,
        "demo": undefined,
        "fillColor": "#ccc",
        "height": 50,
        "max": 100,
        "min": 100,
        "onDecrement": [Function],
        "onIncrement": [Function],
        "progress": 0,
        "transition": undefined,
        "width": 400,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ProgressBar
          bgColor={undefined}
          borderRadius={undefined}
          borderStyle={undefined}
          demo={undefined}
          fillColor="#ccc"
          height={50}
          max={100}
          min={100}
          onDecrement={[Function]}
          onIncrement={[Function]}
          progress={0}
          transition={undefined}
          width={400}
/>,
        "className": "progress-bar-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "bgColor": undefined,
          "borderRadius": undefined,
          "borderStyle": undefined,
          "demo": undefined,
          "fillColor": "#ccc",
          "height": 50,
          "max": 100,
          "min": 100,
          "onDecrement": [Function],
          "onIncrement": [Function],
          "progress": 0,
          "transition": undefined,
          "width": 400,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
